### ISSUE TITLE
`namedtuple.__replace__` should return `Self`

### ISSUE THREAD

**Issue by max-muoto** [python/mypy#17497]
`__replace__` was added onto `NamedTuple` [as a part of 3.13 related changes](https://github.com/python/mypy/pull/17259), however, instead of returning `None`, it should return `Self`. [Looking in cpython we can see this is the case](https://github.com/python/cpython/blob/53e12025cd7d7ee46ce10cc8f1b722c55716b892/Lib/collections/__init__.py#L498).

Furthermore, this is important for compatibility with `copy.replace`. In https://github.com/python/typeshed/pull/12262 we're adding a protocol that requires replaceable objects implement:

```python
class _SupportsReplace(Protocol):
    def __replace__(self, *args: Any, **kwargs: Any) -> Self: ...
```

With `namedtuple.__replace__` returning `None`, users will return into false positives here, when in reality it should work just fine.

**Comment by ilevkivskyi** [comment]
Do you have an actual test case that fails? Also maybe read a comment on `add_method()` that says
```
            ret: Type | None,  # None means use (patched) self-type
```

**Comment by max-muoto** [comment]
> Do you have an actual test case that fails? Also maybe read a comment on `add_method()` that says
> 
> ```
>             ret: Type | None,  # None means use (patched) self-type
> ```

Apologies, didn't realize that would happen by default - I think this is fine to close out in that case.
